.editorconfig

```diff
@@ -0,0 +1,19 @@
+# https://EditorConfig.org
```

Comment:
I'm not sure if we want to add this to the repo (first time I'm hearing about this file tbh).

Would it be good enough for now to add `.editorconfig` to `.gitignore` assuming you use this for configurations?

Reply:
The whole point is to not have this file in `.gitignore`. It's a way for the repo to force some settings on the editor in a cross-editor manner. So, we want to have this file present when the user clones the repo.

You write code in VSCode, I type it in Vim, someone else is using CLion, etc. Yet, our editors agree to use the same character encoding, insert a final new line at the end of every file, not to leave whitespace at the end of the line, etc. It is language agnostic and editor agnostic. It doesn't give you any guidelines other than that. It does require a plugin for some editors (notably both Vim and VSCode), but without a plugin this file doesn't do any harm either.

EditorConfig files are very common, if you look around GitHub you'll see them everywhere. In fact, they are so popular some tools like Prettier decided to use them as a substitution for some of their own setting.

I could put all prettier settings in it's own file, and this way the only time, say, the final new line would be added to a file is during the `npm run fix`. But if I split these basic things into `.editorconfig` then my editor would do that cleanup on file save, too. It's a question of what we prefer.

Reply:
Alright. It's mainly that I am a bit wary of adding new files I've never heard of to the toplevel. But sure, let's add this and see.

> EditorConfig files are very common, if you look around GitHub you'll see them everywhere.

This one surprises me though since as I've said I've never seen this before. (Not that I am doubting you here)